core/trino-parser/src/test/java/io/trino/sql/parser/TestSqlParser.java

```diff
@@ -873,493 +987,608 @@ public void testFormat()
+        assertThat(statement("SHOW TABLES FROM \"awesome schema\"")).isEqualTo(new ShowTables(location(1, 1), Optional.of(QualifiedName.of(ImmutableList.of(new Identifier(location(1, 18), "awesome schema", true)))), Optional.empty(), Optional.empty()));
+        assertThat(statement("SHOW TABLES IN a LIKE '%$_%' ESCAPE '$'")).isEqualTo(new ShowTables(location(1, 1), Optional.of(QualifiedName.of(ImmutableList.of(new Identifier(location(1, 16), "a", false)))), Optional.of("%$_%"), Optional.of("$")));
     }
 
     @Test
     public void testShowColumns()
     {
-        assertStatement("SHOW COLUMNS FROM a", new ShowColumns(QualifiedName.of("a"), Optional.empty(), Optional.empty()));
-        assertStatement("SHOW COLUMNS FROM a.b", new ShowColumns(QualifiedName.of("a", "b"), Optional.empty(), Optional.empty()));
-        assertStatement("SHOW COLUMNS FROM \"awesome table\"", new ShowColumns(QualifiedName.of("awesome table"), Optional.empty(), Optional.empty()));
-        assertStatement("SHOW COLUMNS FROM \"awesome schema\".\"awesome table\"", new ShowColumns(QualifiedName.of("awesome schema", "awesome table"), Optional.empty(), Optional.empty()));
-        assertStatement("SHOW COLUMNS FROM a.b LIKE '%$_%' ESCAPE '$'", new ShowColumns(QualifiedName.of("a", "b"), Optional.of("%$_%"), Optional.of("$")));
-        assertInvalidStatement("SHOW COLUMNS FROM a.b LIKE null", "mismatched input 'null'. Expecting: <string>");
-        assertInvalidStatement("SHOW COLUMNS FROM a.b LIKE 'a' ESCAPE null'", "mismatched input 'null'. Expecting: <string>");
+        assertThat(statement("SHOW COLUMNS FROM a")).isEqualTo(new ShowColumns(location(1, 1), QualifiedName.of(ImmutableList.of(new Identifier(location(1, 19), "a", false))), Optional.empty(), Optional.empty()));
+        assertThat(statement("SHOW COLUMNS FROM a.b")).isEqualTo(new ShowColumns(location(1, 1), QualifiedName.of(ImmutableList.of(new Identifier(location(1, 19), "a", false), new Identifier(location(1, 21), "b", false))), Optional.empty(), Optional.empty()));
+        assertThat(statement("SHOW COLUMNS FROM \"awesome table\"")).isEqualTo(new ShowColumns(location(1, 1), QualifiedName.of(ImmutableList.of(new Identifier(location(1, 19), "awesome table", true))), Optional.empty(), Optional.empty()));
+        assertThat(statement("SHOW COLUMNS FROM \"awesome schema\".\"awesome table\"")).isEqualTo(new ShowColumns(location(1, 1), QualifiedName.of(ImmutableList.of(new Identifier(location(1, 19), "awesome schema", true), new Identifier(location(1, 36), "awesome table", true))), Optional.empty(), Optional.empty()));
+        assertThat(statement("SHOW COLUMNS FROM a.b LIKE '%$_%' ESCAPE '$'")).isEqualTo(new ShowColumns(location(1, 1), QualifiedName.of(ImmutableList.of(new Identifier(location(1, 19), "a", false), new Identifier(location(1, 21), "b", false))), Optional.of("%$_%"), Optional.of("$")));
+
+        assertThatThrownBy(() -> statement("SHOW COLUMNS FROM a.b LIKE null"))
```

Comment:
Instead of nested assertion, you can follow this pattern. It is easier to read.
This is one example from the `testRowPattern()` method:
```
assertThatThrownBy(() -> SQL_PARSER.createRowPattern("A!"))
                .isInstanceOf(ParsingException.class)
                .hasMessageMatching("line 1:2: mismatched input '!'.*");
```